app/src/main/java/protect/card_locker/OpenWebLinkHandler.java

```diff
@@ -0,0 +1,29 @@
+package protect.card_locker;
+
+import android.content.ActivityNotFoundException;
+import android.content.Intent;
+import android.net.Uri;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+public class OpenWebLinkHandler {
+
+    private static final String TAG = "Catima";
+
+    public void open(AppCompatActivity activity, String url) {
+        if (url == null) {
+            return;
+        }
+
+        Intent intent = new Intent(Intent.ACTION_VIEW);
+        intent.setData(Uri.parse(url));
+        try {
+            activity.startActivity(intent);
+        } catch (ActivityNotFoundException e) {
+            Toast.makeText(activity, R.string.failedToOpenUrl, Toast.LENGTH_LONG).show();
+            Log.e(TAG, "No activity found to handle intent", e);
+        }
+    }
```

Comment:
I think it would be cleaner to just make this an `openBrowser(url)` function that can throw the `ActivityNotFoundException`. That way it doesn't need to have the activity and the `openBrowser` function can be re-used in other classes in the future.

Reply:
You will still need the activity to launch the intent

Reply:
```
(new OpenWebLinkHandler()).open(.)

(new OpenWebLinkHandler()).openBrowser(.)
```
I renamed it, but I guess it was not necessary.

Reply:
Ah oops, I totally misread that bit of code.